forch/forch_metrics.py

```diff
@@ -0,0 +1,63 @@
+"""Module to expose varz interface"""
+
+import functools
+import logging
+import threading
+from prometheus_client import Info, generate_latest, REGISTRY
+
+import forch.http_server
+
+LOGGER = logging.getLogger('metrics')
+
+class ForchMetrics():
+    """Class that implements the module that exposes varz for metrics"""
+    _reg = REGISTRY
+    def __init__(self, local_port, config):
+        self._local_port = local_port
+        self._config = config
+        self._http_server = None
+        self._metrics = {}
+
+    def start(self):
+        """Start serving varz"""
+        self._add_vars()
+        self._http_server = forch.http_server.HttpServer(self._config, self._local_port)
+        try:
+            self._http_server.map_request('', self.get_metrics)
+        except Exception as e:
+            self._http_server.map_request('', functools.partial(self._show_error, e))
+        finally:
+            threading.Thread(target=self._http_server.start_server(), daemon=True).start()
+
+    def stop(self):
+        """Kill varz server"""
+        self._http_server.stop_server()
+
+    def update_var(self, var, value):
+        """Update given varz with new value"""
+        varz = self._metrics.get(var)
+        if not varz:
+            LOGGER.error('Error updating to varz %s since it is not known.', var)
```

Comment:
seems like this should be a hard error (throw exception)

Reply:
Done